[v2] Modules located outside of the project's scope throw an error upon importing

<!--
  To make it easier for us to help you — please follow the suggested format below.

  Useful Links:
  - Documentation: https://www.gatsbyjs.org/docs/
  - How to File an Issue: https://www.gatsbyjs.org/docs/how-to-file-an-issue/

  Before opening a new issue, please search existing issues https://github.com/gatsbyjs/gatsby/issues
-->

## Description

Since gatsby@2.0.0-alpha.58, I cannot import local npm packages from a path relative to the project but outside its scope (e.g. in `gatsby-project/src/index.js`, `import '../../other-project'` throws an error).

It seems that this bug is **not** present in gatsby@2.0.0-alpha.56, and there hasn't been a gatsby@2.0.0-alpha.57 release on npm.

### Steps to reproduce

Use ES2015+ syntax in a package outside of a Gatsby project's scope and import that package using relative paths.

### Actual result

_In the following example, `untitled-project` is an npm package with its sources lying in `untitled-project/src`, imported by `untitled-project/visualizer` (a Gatsby project) using `import something from '../../../src'`._

```
D:\Development\untitled-project\visualizer>yarn develop
yarn run v1.7.0
$ gatsby develop
success delete html files from previous builds — 0.004 s
success open and validate gatsby-config — 0.004 s
success copy gatsby files — 0.043 s
success onPreBootstrap — 0.095 s
success source and transform nodes — 0.015 s
success building schema — 0.081 s
success createPages — 0.000 s
success createPagesStatefully — 0.022 s
success onPreExtractQueries — 0.000 s
success update schema — 0.038 s
success extract queries from components — 0.040 s
success run graphql queries — 0.018 s
success write out page data — 0.006 s
success write out redirect data — 0.001 s
success onPostBootstrap — 0.000 s
 
info bootstrap finished - 4.704 s
 
 ERROR  Failed to compile with 3 errors                                                                         10:56:45
 
These dependencies were not found:
 
* core-js/modules/es6.math.sign in ../src/az-el.js
* core-js/modules/es6.object.assign in ../src/az-el.js
* react-hot-loader in ../src/az-el.js
 
To install them, you can run: npm install --save core-js/modules/es6.math.sign core-js/modules/es6.object.assign react-hot-loader
× ｢wdm｣:
ERROR in ../src/az-el.js
Module not found: Error: Can't resolve 'core-js/modules/es6.math.sign' in 'D:\Development\untitled-project\src'
 @ ../src/az-el.js 1:0-40
 @ ../src/index.js
 @ ./src/pages/index.jsx
 @ ./.cache/sync-requires.js
 @ ./.cache/root.js
 @ ./.cache/app.js
 @ multi ./node_modules/react-hot-loader/patch.js (webpack)-hot-middleware/client.js?path=http://localhost:8000/__webpack_hmr&reload=true&overlay=false D://Development//untitled-project//visualizer//.cache//app
 
ERROR in ../src/az-el.js
Module not found: Error: Can't resolve 'core-js/modules/es6.object.assign' in 'D:\Development\untitled-project\src'
 @ ../src/az-el.js 3:0-44
 @ ../src/index.js
 @ ./src/pages/index.jsx
 @ ./.cache/sync-requires.js
 @ ./.cache/root.js
 @ ./.cache/app.js
 @ multi ./node_modules/react-hot-loader/patch.js (webpack)-hot-middleware/client.js?path=http://localhost:8000/__webpack_hmr&reload=true&overlay=false D://Development//untitled-project//visualizer//.cache//app
 
ERROR in ../src/az-el.js
Module not found: Error: Can't resolve 'react-hot-loader' in 'D:\Development\untitled-project\src'
 @ ../src/az-el.js 6:20-47 216:23-50 218:20-47
 @ ../src/index.js
 @ ./src/pages/index.jsx
 @ ./.cache/sync-requires.js
 @ ./.cache/root.js
 @ ./.cache/app.js
 @ multi ./node_modules/react-hot-loader/patch.js (webpack)-hot-middleware/client.js?path=http://localhost:8000/__webpack_hmr&reload=true&overlay=false D://Development//untitled-project//visualizer//.cache//app
i ｢wdm｣: Failed to compile.
```

### Environment

- Gatsby version (npm list gatsby): 2.0.0-alpha.58
- gatsby-cli version (gatsby --version): 2.0.0-alpha.58
- Node.js version: 9.11.2
- Operating System: Windows 10 x64

--- a/packages/gatsby/src/utils/webpack.config.js
+++ b/packages/gatsby/src/utils/webpack.config.js
@@ -370,9 +370,9 @@ module.exports = async (
       ],
       alias: {
         gatsby$: directoryPath(path.join(`.cache`, `gatsby-browser-entry.js`)),
-        'core-js': path.dirname(require.resolve('core-js/package.json')),
-        'react-hot-loader': path.dirname(
-          require.resolve('react-hot-loader/package.json'),
+        `core-js`: path.dirname(require.resolve(`core-js/package.json`)),
+        `react-hot-loader`: path.dirname(
+          require.resolve(`react-hot-loader/package.json`),
         ),
       },
     }

--- a/packages/gatsby/src/utils/webpack.config.js
+++ b/packages/gatsby/src/utils/webpack.config.js
@@ -370,6 +370,10 @@ module.exports = async (
       ],
       alias: {
         gatsby$: directoryPath(path.join(`.cache`, `gatsby-browser-entry.js`)),
+        'core-js': path.dirname(require.resolve('core-js/package.json')),
+        'react-hot-loader': path.dirname(
+          require.resolve('react-hot-loader/package.json'),
+        ),
       },
     }
   }

--- a/packages/gatsby/src/utils/webpack.config.js
+++ b/packages/gatsby/src/utils/webpack.config.js
@@ -370,6 +370,9 @@ module.exports = async (
       ],
       alias: {
         gatsby$: directoryPath(path.join(`.cache`, `gatsby-browser-entry.js`)),
+        // Using directories for module resolution is mandatory because
+        // relative path imports are used sometimes
+        // See https://stackoverflow.com/a/49455609/6420957 for more details
         'core-js': path.dirname(require.resolve(`core-js/package.json`)),
         'react-hot-loader': path.dirname(
           require.resolve(`react-hot-loader/package.json`),

--- a/packages/gatsby/src/utils/webpack.config.js
+++ b/packages/gatsby/src/utils/webpack.config.js
@@ -370,8 +370,8 @@ module.exports = async (
       ],
       alias: {
         gatsby$: directoryPath(path.join(`.cache`, `gatsby-browser-entry.js`)),
-        `core-js`: path.dirname(require.resolve(`core-js/package.json`)),
-        `react-hot-loader`: path.dirname(
+        'core-js': path.dirname(require.resolve(`core-js/package.json`)),
+        'react-hot-loader': path.dirname(
           require.resolve(`react-hot-loader/package.json`),
         ),
       },

